### PR TITLE
[Starter] [Core] Modify connector v2 starter

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Purpose of this pull request
Modify connector v2 starter and remove spark/flink in seatunnel-core-starter
SubTask for #2690 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/config/FlinkJobType.java [hunk 1]
```diff
@@ -18,9 +18,7 @@
 package org.apache.seatunnel.core.starter.flink.config;
 
 public enum FlinkJobType {
-    JAR("start-seatunnel-flink.sh"),
-    SQL("start-seatunnel-sql.sh"),
-    ;
+    JAR("start-seatunnel-flink-connector-v2.sh");
```
Comment (ashulin) [inline seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/config/FlinkJobType.java]: Is this still used? flink-starter will only start this shell

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/config/ExecutionFactory.java [hunk 2]
```diff
@@ -1,84 +0,0 @@
...
-                        break;
-                    default:
-                        execution = new FlinkBatchExecution(flinkEnvironment);
-                }
-                break;
-            default:
-                throw new IllegalArgumentException("No suitable engine");
-        }
-        LOGGER.info("current execution is [{}]", execution.getClass().getName());
-        return (Execution<BaseSource<ENVIRONMENT>, BaseTransform<ENVIRONMENT>, BaseSink<ENVIRONMENT>, ENVIRONMENT>) execution;
```
Comment (ashulin) [inline seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/config/ExecutionFactory.java:81]: Where did this part of the logic change go?

File: seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/config/FlinkJobType.java (hunk 1 above)
Comment (Hisoka-X) [reply seatunnel-core/seatunnel-flink-starter/src/main/java/org/apache/seatunnel/core/starter/flink/config/FlinkJobType.java]: Yes, I will remove it.

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/config/ExecutionFactory.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/config/ExecutionFactory.java:81]: This Factory never be used in connector-v2, same logic in FlinkEnvironment/SparkEnvironment.

Comment (ashulin) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/config/ExecutionFactory.java:81]: ok

**Review by hailin0 — DISMISSED** [review]

**Review by ashulin — DISMISSED** [review]

**Comment by Hisoka-X** [comment]
@ashulin @hailin0 Hi, Please check again, thanks!

**Review by ashulin — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]
+1

**Pull request merged**